src/resultspagediff.cpp

```diff
@@ -0,0 +1,269 @@
+            &ResultsPageDiff::navigateToCode);
+    connect(m_resultsCallerCalleePage, &ResultsCallerCalleePage::navigateToCodeFailed, this,
+            &ResultsPageDiff::showError);
+    connect(m_resultsCallerCalleePage, &ResultsCallerCalleePage::selectSymbol, m_timeLineWidget,
+            &TimeLineWidget::selectSymbol);
+
+    connect(m_resultsBottomUpPage, &ResultsBottomUpPage::jumpToCallerCallee, this,
+            &ResultsPageDiff::onJumpToCallerCallee);
+    connect(m_resultsBottomUpPage, &ResultsBottomUpPage::openEditor, this, &ResultsPageDiff::onOpenEditor);
+    connect(m_resultsBottomUpPage, &ResultsBottomUpPage::selectSymbol, m_timeLineWidget, &TimeLineWidget::selectSymbol);
+    connect(m_resultsTopDownPage, &ResultsTopDownPage::jumpToCallerCallee, this,
+            &ResultsPageDiff::onJumpToCallerCallee);
+    connect(m_resultsTopDownPage, &ResultsTopDownPage::openEditor, this, &ResultsPageDiff::onOpenEditor);
+    connect(m_resultsTopDownPage, &ResultsTopDownPage::selectSymbol, m_timeLineWidget, &TimeLineWidget::selectSymbol);
+    connect(m_resultsFlameGraphPage, &ResultsFlameGraphPage::jumpToCallerCallee, this,
+            &ResultsPageDiff::onJumpToCallerCallee);
+    connect(m_resultsFlameGraphPage, &ResultsFlameGraphPage::openEditor, this, &ResultsPageDiff::onOpenEditor);
+    connect(m_resultsFlameGraphPage, &ResultsFlameGraphPage::selectStack, m_timeLineWidget,
+            &TimeLineWidget::selectStack);
+    connect(m_timeLineWidget, &TimeLineWidget::stacksHovered, m_resultsFlameGraphPage,
+            &ResultsFlameGraphPage::setHoveredStacks);
+
+    connect(m_fileA, &PerfParser::parsingStarted, this, [this]() {
+        // disable when we apply a filter
+        m_contents->setEnabled(false);
+        repositionFilterBusyIndicator();
+        m_filterBusyIndicator->setVisible(true);
+    });
+    connect(m_fileA, &PerfParser::parsingFinished, this, [this]() {
+        // re-enable when we finished filtering
+        m_contents->setEnabled(true);
+        m_filterBusyIndicator->setVisible(false);
+    });
+
+    for (const auto& parser : {m_fileA, m_fileB}) {
+        connect(parser, &PerfParser::parsingStarted, this, [this] { m_runningParsersCounter++; });
+        // TODO: error handling
+        connect(parser, &PerfParser::parsingFailed, this, [this] { m_runningParsersCounter--; });
+        connect(parser, &PerfParser::parsingFinished, this, [this] {
+            m_runningParsersCounter--;
+            if (m_runningParsersCounter == 0) {
+                emit parsingFinished();
+            }
+        });
+    }
+
+    connect(this, &ResultsPageDiff::parsingFinished, this, [this] {
+        const auto bottomUpData =
+            Data::BottomUpResults::diffBottomUpResults(m_fileA->bottomUpResults(), m_fileB->bottomUpResults());
+        m_resultsBottomUpPage->setBottomUpResults(bottomUpData);
+    });
+
+    {
+        // create a busy indicator
```

Comment:
this (and other areas of this file?) looks like it's copy'n'pasted. please share the code by creating utility functions as needed